src/middlewared/middlewared/schema.py

```diff
@@ -178,17 +178,24 @@ class IPAddr(Str):
 
     def __init__(self, *args, **kwargs):
         self.cidr = kwargs.pop('cidr', False)
+        self.cidr_strict = kwargs.pop('cidr_strict', False)
         super(IPAddr, self).__init__(*args, **kwargs)
 
     def validate(self, value):
+        verrors = ValidationErrors()
+
         if value:
             try:
                 if self.cidr:
-                    ipaddress.ip_interface(value)
+                    ipaddress.ip_network(value, strict=self.cidr_strict)
```

Comment:
We should add this to the afp and smb plugins as well, otherwise I believe they will fail, which is what that change was originally intended for.

Reply:
@skarekrow add what? By the way, why was it 'ip_interface' (it also accepts just IP address w/o mask which I think is not what is needed)? Also 'strict' is 'False' by default, are networks with host bits set valid for AFP and SMB?